### PR TITLE
Less fatal logging

### PR DESCRIPTION
**Pull request by cavis**
Stop logging error statuses < 500 as fatal.  In fact, stop logging them at all!

### CODE REVIEW

File: lib/hal_api/controller/exceptions.rb [hunk 1]
```diff
@@ -7,22 +7,27 @@ module HalApi::Controller::Exceptions
...
       end
-    end
 
-    notice_error(exception) if error.status >= 500
+    error =
+      if exception.is_a?(HalApi::Errors::ApiError) && exception.status >= 500
+        log_error(request.env, wrapper)
+        notice_error(exception)
+        exception
+      elsif exception.is_a?(HalApi::Errors::ApiError)
```
Comment (kookster) [inline lib/hal_api/controller/exceptions.rb:22]: So this is a non-5xx error, like a 404 or 401 - those won't get logged at all, not even to info or debug?

Comment (kookster) [reply lib/hal_api/controller/exceptions.rb:22]: I think I would still log these just not as `fatal`, maybe pass in a severity to the `log_error` method?

**Review by kookster — APPROVED** [review]
I kinda think it ought to still log on 400s, but maybe those will be in the regular request/response log lines from rails, so it's fine? I think it's fine. Approve!

File: lib/hal_api/controller/exceptions.rb (hunk 1 above)
Comment (cavis) [reply lib/hal_api/controller/exceptions.rb:22]: Yeah, I thought that a first too.  But turns out their statuses get logged by the regular rails request logger - so you see a response going out with status=401 (or whatever code).  And in the cases I checked, there was no additional info gained by `logger.warn`-ing the error itself.

**Pull request merged**